osu.Game/Beatmaps/BeatmapUpdaterMetadataLookup.cs

```diff
@@ -85,11 +85,11 @@ public void Update(BeatmapSetInfo beatmapSet, bool preferOnlineFetch)
 
         private bool shouldDiscardLookupResult(OnlineBeatmapMetadata result, BeatmapInfo beatmapInfo)
         {
-            if (beatmapInfo.OnlineID > 0 && result.BeatmapID != beatmapInfo.OnlineID)
-            {
-                Logger.Log($"Discarding metadata lookup result due to mismatching online ID (expected: {beatmapInfo.OnlineID} actual: {result.BeatmapID})", LoggingTarget.Database);
-                return true;
-            }
+            // previously this used to check whether the `OnlineID` of the looked-up beatmap matches the local `OnlineID`.
+            // unfortunately it appears that historically stable mappers would apply crude hacks to fix unspecified "issues" with submission
+            // which would amount to reusing online IDs of other beatmaps.
+            // this means that the online ID in the `.osu` file is not reliable, and this cannot be fixed server-side
+            // because updating the maps retroactively would break stable (by losing all of users' local scores).
 
             if (beatmapInfo.OnlineID == -1 && result.MD5Hash != beatmapInfo.MD5Hash)
```

Comment:
On a quick read of the code, I'm not sure that we can ever hit this scenario anymore. The code run prior to this is a `tryLookup` which is based on md5 matching alone, so at this point it should be guaranteed that the MD5 matches.

I may be wrong, but we can probably do away with this method altogether?

Reply:
Good point, and the assertion holds up as far as I can tell. Removed in https://github.com/ppy/osu/commit/2b0fd3558f2f7f1c6a79dbba62ab80f40d656ca9.